crates/nebula-authorization/src/database/workspace_migration/m20241128_001_init_authorization.rs

```diff
@@ -0,0 +1,116 @@
+use async_trait::async_trait;
+use sea_orm_migration::{prelude::*, schema::*};
+
+#[derive(DeriveIden)]
+pub enum MachineIdentity {
+    Table,
+    Id,
+    Label,
+    OwnerGid,
+    CreatedAt,
+    UpdatedAt,
+}
+
+#[derive(DeriveIden)]
+pub enum MachineIdentityAttribute {
+    Table,
+    Id,
+    MachineIdentityId,
+    Key,
+    Value,
+    CreatedAt,
+    UpdatedAt,
+}
+
+#[derive(DeriveIden)]
+pub enum MachineIdentityToken {
+    Table,
+    Id,
+    MachineIdentityId,
+    Token,
+    CreatedAt,
+    UpdatedAt,
+}
+
+#[derive(DeriveMigrationName)]
+pub struct Migration;
+
+#[async_trait]
+impl MigrationTrait for Migration {
+    async fn up(&self, manager: &SchemaManager) -> Result<(), DbErr> {
+        manager
+            .create_table(
+                Table::create()
+                    .table(MachineIdentity::Table)
+                    .col(char_len(MachineIdentity::Id, 26).primary_key())
+                    .col(string_len(MachineIdentity::Label, 255))
+                    .col(string_len(MachineIdentity::OwnerGid, 255))
```

Comment:
I suggest removing the length restriction (255) on the OwnerGid column. Having a fixed length limitation could potentially create security vulnerabilities if the GID exceeds this limit, leading to truncation or rejection of valid identifiers.

Reply:
Strings with unlimited length do not exist in SQL, and if very long strings are used, they cannot be indexed. It would be reasonable to impose a length restriction on the GID.

Reply:
According to the PostgreSQL documentation (https://www.postgresql.org/docs/current/datatype-character.html#DATATYPE-CHARACTER), PostgreSQL does provide the `text` data type.   
Shouldn't we use it in this case?

Reply:
The following reasons make it unsuitable for use as an identifier:  
- When used as an index, it may cause errors that are difficult to identify due to index length limitations.  
- It is stored in a different space than typical data, leading to additional resource consumption.